libs/analytics/src/widget/orderLifecycleAnalytics.ts

```diff
@@ -0,0 +1,243 @@
+      sellAmount: payload.inputAmount?.toString() || '',
+      buyAmount: payload.outputAmount?.toString() || '',
+    }
+  }
+
+  if (isOrderPayload(payload)) {
+    return {
+      walletAddress: safeGetString(payload.order, 'owner'),
+      orderId: safeGetString(payload.order, 'uid'),
+      chainId: payload.chainId,
+      sellToken: safeGetString(payload.order, 'sellToken'),
+      buyToken: safeGetString(payload.order, 'buyToken'),
+      sellAmount: safeGetString(payload.order, 'sellAmount'),
+      buyAmount: safeGetString(payload.order, 'buyAmount'),
+      ...(payload.transactionHash ? { transactionHash: payload.transactionHash } : {}),
+    }
+  }
+
+  // Fallback for unexpected payload types
+  return createDefaultTrackingPayload()
+}
+
+// Helper to set up an event handler with common logic
+export const setupEventHandler = (
+  analytics: CowAnalytics,
+  event: CowWidgetEvents,
+  payload: unknown,
+  getAdditionalProps: (payload: any) => Record<string, any>,
+  isPostedOrder = false,
+) => {
+  const eventName = EVENT_MAPPING[event]
+  if (eventName) {
+    const commonProps = getCommonOrderProperties(payload, isPostedOrder)
+    const additionalProps = getAdditionalProps(payload)
+
+    analytics.sendEvent(eventName, {
+      ...commonProps,
+      ...additionalProps,
+    })
+  } else {
+    console.warn('No event mapping found for event:', event)
+  }
+}
+
+// Event handler setup functions
+export const handlePostedOrder = (analytics: CowAnalytics, payload: unknown) => {
+  setupEventHandler(
+    analytics,
+    CowWidgetEvents.ON_POSTED_ORDER,
+    payload,
+    (p) => ({
+      sellTokenSymbol: safeGetString(p.inputToken, 'symbol'),
+      buyTokenSymbol: safeGetString(p.outputToken, 'symbol'),
+      orderType: p.orderType,
+      kind: p.kind,
+      receiver: p.receiver || '',
+      orderCreationHash: p.orderCreationHash || '',
+    }),
+    true,
+  )
+}
+
+export const handleFulfilledOrder = (analytics: CowAnalytics, payload: unknown) => {
+  setupEventHandler(analytics, CowWidgetEvents.ON_FULFILLED_ORDER, payload, (p) => ({
+    executedSellAmount: safeGetString(p.order, 'executedSellAmount'),
+    executedBuyAmount: safeGetString(p.order, 'executedBuyAmount'),
+    executedFeeAmount: safeGetString(p.order, 'executedFeeAmount'),
+  }))
+}
+
+export const handleCancelledOrder = (analytics: CowAnalytics, payload: unknown) => {
+  setupEventHandler(analytics, CowWidgetEvents.ON_CANCELLED_ORDER, payload, (p) => ({
+    reason: 'cancelled',
+    transactionHash: p.transactionHash || '',
+  }))
+}
+
+export const handleExpiredOrder = (analytics: CowAnalytics, payload: unknown) => {
+  setupEventHandler(analytics, CowWidgetEvents.ON_EXPIRED_ORDER, payload, () => ({
+    reason: 'expired',
+  }))
+}
+
+export const setupEventHandlers = (
+  eventEmitter: SimpleCowEventEmitter<CowWidgetEventPayloadMap, CowWidgetEvents>,
+  getAnalytics: () => CowAnalytics | undefined,
```

Comment:
Since this is within the analytics lib, does it still make sense to take this fn as a parameter?
Do you expect to have different implementations of it according to where it's called?
Otherwise that's what I suggested before, simply import the fn into this module, without taking it as a parameter.